src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -206,32 +252,58 @@ static SyntaxNode TryGetAcceptedNodeOrExtracted(SyntaxNode node, Predicate<Synta
                     return node;
                 }
 
-                var extractedNode = extractNode(node, syntaxFacts);
+                var extractedNode = extractNode(node, syntaxFacts, extractParentsOfHeader);
                 return (extractedNode != null && predicate(extractedNode))
                     ? extractedNode
                     : null;
             }
         }
 
+        private static TextSpan GetSpanWithoutAttributes(SyntaxNode node, SyntaxNode root, ISyntaxFactsService syntaxFacts)
+        {
+            // Span without AttributeLists
+            // - No AttributeLists -> original .Span
+            // - Some AttributeLists -> (first non-trivia/comment Token.Span.Begin, original.Span.End)
+            //   - We need to be mindful about comments due to:
+            //      // [Test1]
+            //      //Comment1
+            //      [||]object Property1 { get; set; }
+            //     the comment node being part of the next token's (`object`) leading trivia and not the AttributeList's node.
+            var attributeList = syntaxFacts.GetAttributeLists(node);
+            if (attributeList.Any())
+            {
+                var endOfAttributeLists = attributeList.Last().Span.End;
+                var afterAttributesToken = root.FindTokenOnRightOfPosition(endOfAttributeLists);
+
+                return TextSpan.FromBounds(afterAttributesToken.Span.Start, node.Span.End);
```

Comment:
note: i've learned from @jnm2 that he commonly does things like this:


```c#
[Foo]
[Bar][|
public void Whatever()
{
}
|]
```

That's because he drags-up on the right to select, and will commonly select up to the end of hte *previous* line of the construct he cares about.

Can we test for this case?  If it works, great.  If not, can we make it work?  Thanks!

Reply:
@CyrusNajmabadi In this case it would be:

```
[Foo]
[Bar][|
public void Whatever()
{
}|]
```

Reply:
@jnm2 Sure :)   We should def make sure both of those work. :)

Reply:
Better example:

```
public class C
{
    public void Foo()
    {
        // ...
    }[|

    public void Bar()
    {
        // ...
    }|]
}
```

<snip>